src/python/tests/fasta/test_fasta_prep_data.py

```diff
@@ -0,0 +1,149 @@
+            input_fasta: Name of the fasta file with example input, in the test folder.
+            input_gbff: Name of the input GBFF example input, in the test folder.
+            expected_output_fasta: Name of the output fasta file with expected output, in the test folder.
+            pep_mode: Boolean flag to set processing in peptide mode.
+        """
+
+        fasta_input_path = self.data_dir / input_fasta
+        if input_gbff is not None:
+            gbff_input_path = self.data_dir / input_gbff
+        else:
+            gbff_input_path = input_gbff
+        fasta_output_path = tmp_path / f"{input_fasta}.test.fasta"
+
+        FastaProcessing.prep_fasta_data(
+            fasta_infile=fasta_input_path,
+            genbank_infile=gbff_input_path,
+            fasta_outfile=fasta_output_path,
+            peptide_mode=pep_mode,
+        )
+
+        expected_path = self.data_dir / expected_output_fasta
+
+        assert filecmp.cmp(fasta_output_path, expected_path)
+
+    @pytest.mark.parametrize(
+        "input_gbff, excluded_seq_region, output, scenario",
+        [
+            (
+                "input.gbff.gz",
+                ["LR605957.1"],
+                ["VWP78966.1", "VWP78967.1", "VWP78968.1"],
+                "Normal",
+            ),
+            (  
+                "input.mod.gbff.gz", 
+                ["LR605957.1"], 
+                ["VWP78966.1", "VWP78967.1", "VWP78968.1"],
+                "Modified"
+            ),
+        ],
+    )
+    def test_exclude_seq_regions(
+        self,
+        input_gbff: str,
+        excluded_seq_region: List[str],
+        output: List[str],
+        scenario: str,
+    ) -> None:
+        """Tests the `process.get_peptides_to_exclude()` function.
+
+        Args:
+            input_gbff: Name of the input GBFF example input, in the test folder.
+            excluded_seq_region: Set of sequence regions to be excluded.
+            output: Set of proteins expected to be excluded given excluded_seq_region
+            scenario: Normal or modified function test.
+        """
+
+        if input_gbff is not None:
+            gbff_input_path = self.data_dir / input_gbff
+        else:
+            gbff_input_path = input_gbff
+
+        seq_region_to_exclude = set(excluded_seq_region)
+
+        # Testing 'Normal' operation i.e. intact and correct GenBank gbff
+        if scenario == "Normal":
+            excluded_proteins = FastaProcessing.get_peptides_to_exclude(
+                gbff_input_path, seq_region_to_exclude
+            )
+            assert set(excluded_proteins) == set(output)  # Expecting 3 peptides on seq_region (LR605957.1)
+        # Improper operation with corrupt GenBank gbff
+        else:
+            pytest.raises(
+                FastaProcessing.FastaParserError,
+                FastaProcessing.get_peptides_to_exclude,
+                gbff_input_path,
+                seq_region_to_exclude,
+            )
```

Comment:
There is a better way to test for exceptions without having to have an `if-then-else` clause:
```suggestion
@pytest.mark.parametrize(
        "input_gbff, excluded_seq_region, output, expectation",
        [
            (
                "input.gbff.gz",
                set(["LR605957.1"]),
                set(["VWP78966.1", "VWP78967.1", "VWP78968.1"]),
                does_not_raise()
            ),
            (  
                "input.mod.gbff.gz",
                set(["LR605957.1"]),
                set(["VWP78966.1", "VWP78967.1", "VWP78968.1"]),
                pytest.raises(FastaProcessing.FastaParserError)
            ),
        ],
    )
    def test_exclude_seq_regions(
        self, input_gbff: str, excluded_seq_regions: Set[str], output: Set[str], expectation: ContextManager,
    ) -> None:
        """Tests the `process.get_peptides_to_exclude()` function.

        Args:
            input_gbff: Name of the input GBFF example input, in the test folder.
            excluded_seq_regions: Set of sequence regions to be excluded.
            output: Set of proteins expected to be excluded given excluded_seq_region
            expectation: Context manager for the expected exception, i.e. the test will only pass if that
                exception is raised. Use `~contextlib.nullcontext` if no exception is expected.

        """
        if input_gbff is not None:
            gbff_input_path = self.data_dir / input_gbff
        else:
            gbff_input_path = input_gbff
        with expectation:
            excluded_proteins = FastaProcessing.get_peptides_to_exclude(gbff_input_path, excluded_seq_regions)
            assert set(excluded_proteins) == set(output)
```